python/utils.py

```diff
@@ -133,6 +133,7 @@ def parfor(func, in_list, out_shape=None, n_jobs=-1, func_args=[],
     """
     if n_jobs == -1:
         n_jobs = multiprocessing.cpu_count()
+        n_jobs=n_jobs-1   
```

Comment:
Great. Might not help with memory issues, but at least it will (hopefully) leave one CPU available to log into the machine and shut things down.